akka-actor-typed/src/main/scala/akka/actor/typed/delivery/internal/ConsumerControllerImpl.scala

```diff
@@ -528,7 +528,7 @@ private class ConsumerControllerImpl[A](
     reverseCollectedChunks.foreach { seqMsg =>
       builder ++= seqMsg.message.asInstanceOf[ChunkedMessage].serialized
     }
-    val bytes = builder.result().toArray
+    val bytes = builder.result().toArrayUnsafe()
```

Comment:
@johanandren Would be nice to be able to get an unsafe-array directly from the builder, to avoid having to go through a new ByteString.

Reply:
`result()` will not create a new ByteString if the size is 0 or 1, and that's the case when `toArrayUnsafe` makes a difference

Reply:
@patriknw Great point. Might even be worth mentioning as a comment!